File: Python3/프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    cnt, left, right = 1, 0, 1
    
    while right < len(s):
        cnt, x = 1, s[left]
        while right < len(s) and cnt != 0:
            cnt += 1 if s[right] == x else -1
            right += 1

        answer += 1
        left, right = right, right + 1
    
    return answer + int(s[left:right] != '')

File: C++/프로그래머스/1/140108. 문자열 나누기/문자열 나누기.cpp
#include <string>
#include <iostream>

using namespace std;

int solution(string s) {
    int answer = 0;
    int left = 0, right = 1, cnt;
    char x;
    
    while(right < s.size()) {
        cnt = 1;
        x = s[left];
        while(right < s.size() && cnt != 0) {
            cnt += (x == s[right] ? 1 : -1);
            right++;
        }
        answer++;
        left = right++;
    }
    
    return answer + int(left < s.size());
}